package.json

```diff
@@ -6,6 +6,24 @@
     "name": "Arjun Vegda",
     "url": "https://arjunvegda.com"
   },
+  "sideEffects": false,
+  "source": "./src/index.ts",
+  "main": "./dist/index.cjs.js",
+  "module": "./dist/index.esm.js",
+  "types": "./dist/index.d.ts",
+  "exports": {
+    ".": {
+      "types": "./dist/index.d.ts",
+      "require": "./dist/index.cjs.js",
+      "default": "./dist/index.esm.js"
+    },
+    "./utils": {
+      "types": "./dist/utils.d.ts",
+      "require": "./dist/utils.cjs.js",
+      "default": "./dist/utils.esm.js"
+    },
+    "./package.json": "./package.json"
+  },
```

Comment:
Can we follow exactly what we have in `jotai`? There are many subtle things. `.mjs` support might be too hard, but maybe required for vite users?

Reply:
Good call! I went through some GH history of `Jotai` and `Zustand` to get context around `.mjs` issue identified in Zustand a while ago. I created a [repro on SlackBlitz](https://stackblitz.com/edit/vitejs-vite-uqylur?file=src%2FDebugger.tsx,src%2FApp.tsx,src%2Fmain.tsx,tsconfig.node.json,tsconfig.json,src%2FCounter.tsx,package.json&terminal=dev) using Vite and use that to test my changes.

Re-ordering the exports for `jotai-devtools` based on this [blog](https://devblogs.microsoft.com/typescript/announcing-typescript-4-7/#package-json-exports-imports-and-self-referencing) seems to load the `esm` types _correctly_ on this [SlackBlitz](https://stackblitz.com/edit/vitejs-vite-uqylur?file=src%2FDebugger.tsx,src%2FApp.tsx,src%2Fmain.tsx,tsconfig.node.json,tsconfig.json,src%2FCounter.tsx,package.json&terminal=dev). Shouldn't it break? I'm not entirely sure _why_ is it working, despite setting `moduleResolution` in `tsconfig.json` to `NodeNext` 🤔 

Creating `.mjs` _with_ `.mts` (TypeScript support) appears to be a bit tricky with `tsup`. I'm not able to generate `.mts` 😞 . I'll continue my investigation.

Reply:
Added support for `.mjs` in [this commit](https://github.com/jotai-labs/jotai-devtools/pull/5#:~:text=Show%20resolved-,build%3A%20add%20support,-8170a30)

**Tested it on SlackBlitz and it automatically picks up `.mts` when using `NodeNext`**
<img width="922" alt="image" src="https://user-images.githubusercontent.com/14841132/211236064-17a5d473-2277-4a3a-9e9f-baa957aae24e.png">

**Tested it on CodeSandbox and it automatically picks up regular `.d.ts` when using `node`**
<img width="688" alt="image" src="https://user-images.githubusercontent.com/14841132/211236157-21c0a358-afb4-4f87-b654-32c7ba616ef6.png">